syntax = "proto3";

message rtc_data {
  optional int32 week = 1;
  optional int32 sec = 2;
  optional int32 min = 3;
  optional int32 hour = 4;
  optional int32 day = 5;
  optional int32 month = 6;
  optional int32 year = 7;
}

message time_range_strategy {
  optional bool is_config = 1;
  optional bool is_enable = 2;
  optional int32 time_mode = 3;
  optional int32 time_data = 4;
  optional rtc_data start_time = 5;
  optional rtc_data stop_time = 6;
}

message plug_ack_message {
  optional int32 ack = 1;
}

message plug_heartbeat_pack {
  optional int32 err_code = 1;
  optional int32 warn_code = 2;
  optional int32 country = 3;
  optional int32 town = 4;
  optional int32 max_cur = 5;
  optional int32 temp = 6;
  optional int32 freq = 7;
  optional int32 current = 8;
  optional int32 volt = 9;
  optional int32 watts = 10;
  optional bool switch = 11;
  optional int32 brightness = 12;
  optional int32 max_watts = 13;
  optional int32 heartbeat_frequency = 14;
  optional bool mesh_enable = 15;
  optional int32 unknown_17 = 17;
  optional int32 unknown_18 = 18;
  optional int32 unknown_19 = 19;
  optional int32 unknown_20 = 20;
  optional int32 unknown_21 = 21;
  optional int32 unknown_22 = 22;
  optional int32 unknown_23 = 23;
  optional int32 unknown_24 = 24;
  optional int32 unknown_25 = 25;
  optional int32 unknown_26 = 26;
  optional int32 unknown_27 = 27;
  optional int32 unknown_28 = 28;
  optional int32 unknown_29 = 29;
  optional int32 unknown_30 = 30;
  optional int32 unknown_31 = 31;
  optional int32 unknown_32 = 32;
  optional int32 unknown_33 = 33;
  optional int32 unknown_34 = 34;
  optional int32 unknown_35 = 35;
  optional int32 unknown_36 = 36;
  optional int32 unknown_37 = 37;
  optional int32 unknown_38 = 38;
}

message plug_switch_message {
  optional int32 plug_switch = 1;
}

message brightness_pack {
  optional int32 brightness = 1;
}

message max_cur_pack {
  optional int32 max_cur = 1;
}

message time_task_config {
  optional string task_name = 1;
  optional time_range_strategy time_range = 2;
  optional int32 type = 3;
}

message time_task_config_post {
  optional int32 index = 1;
  optional time_task_config task_config = 2;
}

message PowerItem {
  optional int64 timestamp = 1;
  optional string timezone = 2;
  optional int32 plug_power = 3;
}

message PowerPack {
  optional int32 sys_seq = 1;
  repeated PowerItem sys_power_stream = 2;
}

message PowerAckPack {
  optional int32 sys_seq = 1;
}

message max_watts_pack {
  optional int32 max_watts = 1;
}

message mesh_ctrl_pack {
  optional int32 mesh_enable = 1;
}

message ret_pack {
  optional bool ret_sta = 1;
}